packages/client/src/publish/ProxyPublisher.ts

```diff
@@ -0,0 +1,39 @@
+import { StreamDefinition } from '../types'
+import { StreamIDBuilder } from '../StreamIDBuilder'
+import { inject, Lifecycle, scoped } from 'tsyringe'
+import BrubeckNode from '../BrubeckNode'
+import { Context } from '../utils/Context'
+
+@scoped(Lifecycle.ContainerScoped)
+export default class ProxyPublisher {
+    constructor(
+        context: Context,
+        private node: BrubeckNode,
+        @inject(StreamIDBuilder) private streamIdBuilder: StreamIDBuilder,
+    ) {
+    }
+
+    async setPublishProxy(streamDefinition: StreamDefinition, nodeId: string): Promise<void> {
+        const streamPartId = await this.streamIdBuilder.toStreamPartID(streamDefinition)
+        await this.node.openPublishProxyConnectionOnStreamPart(streamPartId, nodeId)
+    }
+
+    async removePublishProxy(streamDefinition: StreamDefinition, nodeId: string): Promise<void> {
+        const streamPartId = await this.streamIdBuilder.toStreamPartID(streamDefinition)
+        await this.node.closePublishProxyConnectionOnStreamPart(streamPartId, nodeId)
+    }
+
+    async setPublishProxies(streamDefinition: StreamDefinition, nodeIds: string[]): Promise<void> {
+        const streamPartId = await this.streamIdBuilder.toStreamPartID(streamDefinition)
+        await Promise.allSettled([
+            ...nodeIds.map((nodeId) => this.node.openPublishProxyConnectionOnStreamPart(streamPartId, nodeId))
+        ])
```

Comment:
`Array.prototype.map()` already returns an array so you shouldn't have to spread?
```suggestion
        await Promise.allSettled(
            nodeIds.map((nodeId) => this.node.openPublishProxyConnectionOnStreamPart(streamPartId, nodeId))
        ])
```